### DISCUSSION TITLE
mjData.xquat vs mjData.xmat

### DISCUSSION THREAD

**Discussion by rohit-kumar-j** [google-deepmind/mujoco#343] [Asking for Help] [answered]
What is the difference between `mjData.xquat` vs `mjData.xmat`?

> mjtNum*   xquat;                // Cartesian orientation of body frame      (nbody x 4)
mjtNum*   xmat;                 // Cartesian orientation of body frame      (nbody x 9)

[ref: mjdata](https://mujoco.readthedocs.io/en/latest/APIreference.html?highlight=mjmodel#mjdata)

**Comment by yuvaltassa** [comment, marked as answer]
The former is a quaternion, the latter is a rotation matrix. They represent the same thing (body orientation) in different ways. You can use `mju_quat2Mat` and `mju_mat2Quat` to convert between the two representations.
